{"deviceId":"ESP32001","error":"Plan executor error during findAndModify :: caused by :: Can't extract geo keys: { _id: ObjectId('688c4bb2bffd66bc19925948'), deviceId: \"ESP32001\", __v: 0, alertSettings: { vibrationEnabled: true, tamperingEnabled: true, lowBatteryEnabled: true, geofenceEnabled: false }, batteryThreshold: 20, createdAt: new Date(1754024878873), description: \"Primary vehicle tracking device\", geofence: { center: { coordinates: [], type: \"Point\" }, enabled: false }, hardware: { model: \"ESP32-WROOM-32\", version: \"1.0\", imei: \"123456789012345\" }, isActive: true, lastSeen: new Date(1754024878870), name: \"Vehicle Tracker 1\", updatedAt: new Date(1754024878873) }  Point must only contain numeric elements, instead got type missing","level":"error","message":"Error registering device:","name":"Vehicle Tracker 1","service":"iot-tracking-server","timestamp":"2025-08-01T05:07:58.925Z"}
{"body":{"alertSettings":{"geofenceEnabled":false,"lowBatteryEnabled":true,"tamperingEnabled":true,"vibrationEnabled":true},"batteryThreshold":20,"description":"Primary vehicle tracking device","deviceId":"ESP32001","hardware":{"imei":"123456789012345","model":"ESP32-WROOM-32","version":"1.0"},"name":"Vehicle Tracker 1"},"deviceId":"ESP32001","ip":"::1","level":"error","message":"Error occurred: Plan executor error during findAndModify :: caused by :: Can't extract geo keys: { _id: ObjectId('688c4bb2bffd66bc19925948'), deviceId: \"ESP32001\", __v: 0, alertSettings: { vibrationEnabled: true, tamperingEnabled: true, lowBatteryEnabled: true, geofenceEnabled: false }, batteryThreshold: 20, createdAt: new Date(1754024878873), description: \"Primary vehicle tracking device\", geofence: { center: { coordinates: [], type: \"Point\" }, enabled: false }, hardware: { model: \"ESP32-WROOM-32\", version: \"1.0\", imei: \"123456789012345\" }, isActive: true, lastSeen: new Date(1754024878870), name: \"Vehicle Tracker 1\", updatedAt: new Date(1754024878873) }  Point must only contain numeric elements, instead got type missing","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"MongoServerError: Plan executor error during findAndModify :: caused by :: Can't extract geo keys: { _id: ObjectId('688c4bb2bffd66bc19925948'), deviceId: \"ESP32001\", __v: 0, alertSettings: { vibrationEnabled: true, tamperingEnabled: true, lowBatteryEnabled: true, geofenceEnabled: false }, batteryThreshold: 20, createdAt: new Date(1754024878873), description: \"Primary vehicle tracking device\", geofence: { center: { coordinates: [], type: \"Point\" }, enabled: false }, hardware: { model: \"ESP32-WROOM-32\", version: \"1.0\", imei: \"123456789012345\" }, isActive: true, lastSeen: new Date(1754024878870), name: \"Vehicle Tracker 1\", updatedAt: new Date(1754024878873) }  Point must only contain numeric elements, instead got type missing\n    at Connection.sendCommand (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:305:27)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async Connection.command (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\cmap\\connection.js:333:26)\n    at async Server.command (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\sdam\\server.js:171:29)\n    at async FindOneAndUpdateOperation.executeCommand (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\operations\\command.js:76:16)\n    at async FindOneAndUpdateOperation.execute (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\operations\\find_and_modify.js:85:24)\n    at async tryOperation (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\operations\\execute_operation.js:207:20)\n    at async executeOperation (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\operations\\execute_operation.js:75:16)\n    at async Collection.findOneAndUpdate (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongodb\\lib\\collection.js:500:16)\n    at async model.Query._findOneAndUpdate (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:3513:13)","timestamp":"2025-08-01T05:07:58.927Z","url":"/api/v1/device/register","userAgent":"PostmanRuntime/7.44.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-08-01T05:15:08.673Z","url":"/api/v1/device/register","userAgent":"curl/8.0.1"}
{"deviceId":"ESP32001","error":"Validation failed: geofence.center.coordinates: Coordinates must contain exactly 2 elements [longitude, latitude]","level":"error","message":"Error registering device:","name":"Vehicle Tracker 1","service":"iot-tracking-server","timestamp":"2025-08-01T05:16:03.953Z"}
{"body":{"alertSettings":{"geofenceEnabled":false,"lowBatteryEnabled":true,"tamperingEnabled":true,"vibrationEnabled":true},"batteryThreshold":20,"description":"Primary vehicle tracking device","deviceId":"ESP32001","hardware":{"imei":"123456789012345","model":"ESP32-WROOM-32","version":"1.0"},"name":"Vehicle Tracker 1"},"deviceId":"ESP32001","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Validation failed: geofence.center.coordinates: Coordinates must contain exactly 2 elements [longitude, latitude]","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"ValidationError: Validation failed: geofence.center.coordinates: Coordinates must contain exactly 2 elements [longitude, latitude]\n    at _done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\helpers\\updateValidators.js:231:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\helpers\\updateValidators.js:207:11\n    at schemaPath.doValidate.updateValidator (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\helpers\\updateValidators.js:165:13)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\schemaType.js:1410:9\n    at process.processTicksAndRejections (node:internal/process/task_queues:85:11)","timestamp":"2025-08-01T05:16:03.955Z","url":"/api/v1/device/register","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 464 (line 25 column 17)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 464 (line 25 column 17)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-08-01T05:25:00.729Z","url":"/api/v1/device/update-data","userAgent":"PostmanRuntime/7.44.1"}
{"body":{},"deviceId":"unknown","ip":"::1","level":"error","message":"Error occurred: Expected ',' or '}' after property value in JSON at position 470 (line 25 column 20)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected ',' or '}' after property value in JSON at position 470 (line 25 column 20)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-08-01T05:26:15.247Z","url":"/api/v1/device/update-data","userAgent":"PostmanRuntime/7.44.1"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at gracefulShutdown (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:190:22)\n    at process.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:214:28)\n    at process.emit (node:events:530:35)","timestamp":"2025-09-05T13:48:18.495Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T04:34:11.705Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at gracefulShutdown (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:193:22)\n    at process.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:217:28)\n    at process.emit (node:events:530:35)","timestamp":"2025-09-06T04:35:34.062Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T04:38:55.152Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T04:39:26.070Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at gracefulShutdown (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:193:22)\n    at process.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:217:28)\n    at process.emit (node:events:530:35)","timestamp":"2025-09-06T04:39:35.965Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T04:40:35.754Z"}
{"level":"error","message":"Send OTP error: errorResponse is not a function","service":"iot-tracking-server","stack":"TypeError: errorResponse is not a function\n    at sendOTP (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\authController.js:43:16)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-09-06T04:49:06.241Z"}
{"level":"error","message":"Unhandled Rejection at:","promise":{},"reason":{},"service":"iot-tracking-server","timestamp":"2025-09-06T04:49:06.243Z"}
{"level":"error","message":"Send OTP error: errorResponse is not a function","service":"iot-tracking-server","stack":"TypeError: errorResponse is not a function\n    at sendOTP (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\authController.js:43:16)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-09-06T04:49:52.467Z"}
{"level":"error","message":"Unhandled Rejection at:","promise":{},"reason":{},"service":"iot-tracking-server","timestamp":"2025-09-06T04:49:52.469Z"}
{"level":"error","message":"Send OTP error: errorResponse is not a function","service":"iot-tracking-server","stack":"TypeError: errorResponse is not a function\n    at sendOTP (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\authController.js:43:16)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)","timestamp":"2025-09-06T04:51:04.193Z"}
{"level":"error","message":"Unhandled Rejection at:","promise":{},"reason":{},"service":"iot-tracking-server","timestamp":"2025-09-06T04:51:04.195Z"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T04:51:59.380Z"}
{"level":"error","message":"Send OTP error: Cannot call `otps.countDocuments()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.","service":"iot-tracking-server","stack":"MongooseError: Cannot call `otps.countDocuments()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.\n    at NativeCollection.<computed> [as countDocuments] (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:237:15)\n    at model.Query._countDocuments (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:2785:34)\n    at model.Query.exec (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:4604:80)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async otpSchema.statics.checkRateLimit (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\models\\OTP.js:157:27)\n    at async sendOTP (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\authController.js:26:28)","timestamp":"2025-09-06T04:52:50.408Z"}
{"level":"error","message":"Unhandled Rejection at:","promise":{},"reason":{},"service":"iot-tracking-server","timestamp":"2025-09-06T04:52:50.411Z"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T04:58:19.209Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T04:58:39.308Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T05:31:13.845Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T05:31:31.102Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T05:31:39.552Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"body":{},"deviceId":"unknown","ip":"::ffff:127.0.0.1","level":"error","message":"Error occurred: Expected property name or '}' in JSON at position 1 (line 1 column 2)","method":"POST","params":{},"query":{},"service":"iot-tracking-server","stack":"SyntaxError: Expected property name or '}' in JSON at position 1 (line 1 column 2)\n    at JSON.parse (<anonymous>)\n    at parse (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\types\\json.js:92:19)\n    at C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\body-parser\\lib\\read.js:128:18\n    at AsyncResource.runInAsyncScope (node:async_hooks:214:14)\n    at invokeCallback (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:238:16)\n    at done (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:227:7)\n    at IncomingMessage.onEnd (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\raw-body\\index.js:287:7)\n    at IncomingMessage.emit (node:events:518:28)\n    at endReadableNT (node:internal/streams/readable:1698:12)\n    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)","timestamp":"2025-09-06T05:36:29.407Z","url":"/api/v1/auth/send-otp","userAgent":"curl/8.0.1"}
{"deviceId":"ESPPrototype01","error":"Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.","level":"error","message":"Error retrieving current location:","service":"iot-tracking-server","timestamp":"2025-09-06T06:22:51.943Z"}
{"body":{},"deviceId":"unknown","ip":"::1","level":"error","message":"Error occurred: Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.","method":"GET","params":{},"query":{},"service":"iot-tracking-server","stack":"MongooseError: Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.\n    at NativeCollection.<computed> [as findOne] (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:237:15)\n    at model.Query._findOne (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:2687:45)\n    at model.Query.exec (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:4604:80)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\deviceController.js:245:20","timestamp":"2025-09-06T06:22:51.946Z","url":"/api/v1/device/ESPPrototype01/web-current","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/139.0.0.0 Safari/537.36"}
{"deviceId":"ESPPrototype01","error":"Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.","level":"error","message":"Error retrieving current location:","service":"iot-tracking-server","timestamp":"2025-09-06T06:22:51.961Z"}
{"body":{},"deviceId":"unknown","ip":"::1","level":"error","message":"Error occurred: Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.","method":"GET","params":{},"query":{},"service":"iot-tracking-server","stack":"MongooseError: Cannot call `devices.findOne()` before initial connection is complete if `bufferCommands = false`. Make sure you `await mongoose.connect()` if you have `bufferCommands = false`.\n    at NativeCollection.<computed> [as findOne] (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\drivers\\node-mongodb-native\\collection.js:237:15)\n    at model.Query._findOne (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:2687:45)\n    at model.Query.exec (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\mongoose\\lib\\query.js:4604:80)\n    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)\n    at async C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\controllers\\deviceController.js:245:20","timestamp":"2025-09-06T06:22:51.963Z","url":"/api/v1/device/ESPPrototype01/web-current","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/139.0.0.0 Safari/537.36"}
{"error":"listen EADDRINUSE: address already in use :::3000","level":"error","message":"Uncaught Exception:","service":"iot-tracking-server","stack":"Error: listen EADDRINUSE: address already in use :::3000\n    at Server.setupListenHandle [as _listen2] (node:net:1939:16)\n    at listenInCluster (node:net:1996:12)\n    at Server.listen (node:net:2101:7)\n    at Function.listen (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\node_modules\\express\\lib\\application.js:635:24)\n    at Object.<anonymous> (C:\\Users\\Hp\\Desktop\\dev\\software\\mohit\\fixpoint-drishti-backend\\src\\app.js:247:20)\n    at Module._compile (node:internal/modules/cjs/loader:1730:14)\n    at Object..js (node:internal/modules/cjs/loader:1895:10)\n    at Module.load (node:internal/modules/cjs/loader:1465:32)\n    at Function._load (node:internal/modules/cjs/loader:1282:12)\n    at TracingChannel.traceSync (node:diagnostics_channel:322:14)","timestamp":"2025-09-06T06:24:43.898Z"}
